KEP of the Week 20260405 (#796)
lwkd/lwkd.github.io: _posts/2026-04-05-update.md

## KEP of the Week

[KEP-5075: DRA Consumable Capacity](https://github.com/kubernetes/enhancements/tree/master/keps/sig-scheduling/5075-dra-consumable-capacity)

This KEP introduces support for consumable capacity in Dynamic Resource Allocation (DRA), enabling multiple independent resource claims to allocate and share portions of the same underlying device. Unlike the traditional exclusive allocation model, this approach allows efficient device sharing across unrelated pods and namespaces while ensuring that total allocated capacity remains within device limits through a scheduler-enforced consumable capacity model.

> The proposal is driven by use cases such as shared network devices via CNI, virtual GPU memory allocation, and other multi-allocatable devices where full device exclusivity is inefficient. It introduces mechanisms for capacity-aware scheduling, per-request capacity requirements, consumed capacity tracking, and safeguards to prevent unintended duplicate allocations within a single claim.

Reviewer: Maybe a little shorter:

```suggestion
This approach supports shared network devices via CNI, virtual GPU memory allocation, and other multi-allocatable devices. It introduces mechanisms for capacity-aware scheduling, per-request capacity requirements, consumed capacity tracking, and safeguards to prevent unintended duplicate allocations within a single claim.
```

Author: yes have shortened it.

> The feature was introduced as an alpha capability in Kubernetes 1.34 and is progressing toward beta (targeted for v1.36), with long-term plans for GA as the API and ecosystem mature, aligning with the broader evolution of DRA toward flexible and efficient device resource management.

Reviewer: Cut the last sentence here, make the rest shorter:

```suggestion
The feature became alpha in Kubernetes 1.34 should become beta in 1.36.
```

Author: yes made this change
